CMakeLists.txt

```diff
@@ -40,17 +59,20 @@ file(GLOB_RECURSE sources CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/source/
 # ---- Create library ----
 
 # Note: for header-only libraries change all PUBLIC flags to INTERFACE and create an interface
-# target: add_library(Greeter INTERFACE) set_target_properties(Greeter PROPERTIES
-# INTERFACE_COMPILE_FEATURES cxx_std_17)
+# target: add_library(Greeter INTERFACE)
+add_library(Greeter)
+target_compile_features(Greeter PUBLIC cxx_std_17)
 
-add_library(Greeter ${headers} ${sources})
-
-set_target_properties(Greeter PROPERTIES CXX_STANDARD 17)
+target_sources(Greeter PRIVATE ${headers} ${sources})
 
 # being a cross-platform target, we enforce standards conformance on MSVC
-target_compile_options(Greeter PUBLIC "$<$<BOOL:${MSVC}>:/permissive->")
+if(MSVC)
+  target_compile_options(Greeter PUBLIC /permissive)
+endif()
```

Comment:
I'm not a fan of the explicit conditional branch here. The previous generator expression should have the same effect and avoids explicit branching in the main configuration.

Reply:
this generator expression is exported with empty value MSCV on unix!

Reply:
Yes, that is the expected behaviour. As the empty string is falsy in CMake, the whole expression `$<$<BOOL:>:/permissive->` should be implicitly evaluated to an empty string by CMake.

Reply:
Ynd that is clean for you?

I was confused as I saw this!

Reply:
I recommend avoiding any explicit branching in configure scripts as it makes them harder to reason about. Imo [CMake generator expressions](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html) are a more elegant solution to rare cases where it is indeed necessary to branch. I do agree that they are not exactly beginner-friendly, however as the purpose of this line is clearly stated and is exceptional I think it's better than encouraging explicit branching in build configurations.

Reply:
done